File: src/main/java/tk/skulk/plugin/core/extensions/entityoverride/listeners/ExplosionDropAllListener.java
package tk.skulk.plugin.core.extensions.entityoverride.listeners;

import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.entity.EntityExplodeEvent;
import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.extensions.entityoverride.EntityOverrideExtension;
import tk.skulk.plugin.wrapper.BaseListener;

public final class ExplosionDropAllListener extends BaseListener<EntityOverrideExtension> {
    public ExplosionDropAllListener(final @NotNull EntityOverrideExtension extension) {
        super(extension);
    }

    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
    public void onEntityExplode(final @NotNull EntityExplodeEvent event) {
        event.setYield(100);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/entityOverride/listeners/ExplosionDropAllListener.kt
package tk.skulk.plugin.core.extensions.entityOverride.listeners

import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.entity.EntityExplodeEvent
import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.entityOverride.EntityOverrideExtension
import tk.skulk.plugin.framework.Listener

class ExplosionDropAllListener(extension: EntityOverrideExtension) : Listener<SNPlugin, EntityOverrideExtension>(
    extension
) {
    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
    fun onEntityExplode(event: EntityExplodeEvent) {
        event.yield = 100F
    }
}
